hawc/apps/animal/actions/model_metadata.py

```diff
@@ -0,0 +1,89 @@
+from pydantic import BaseModel
+from rest_framework import serializers
+
+from hawc.apps.animal import models
+from hawc.apps.assessment.models import DoseUnits, Species, Strain
+from hawc.apps.common.actions import BaseApiAction
+
+
+def tuple_to_dict(tuple):
+    return {e[0]: e[1] for e in tuple}
+
+
+class SpeciesSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Species
+        fields = ("id", "name")
+
+
+class StrainSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Strain
+        fields = ("id", "species_id", "name")
+
+
+class DoseUnitsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DoseUnits
+        fields = ("id", "name")
+
+
+class NoInput(BaseModel):
+    pass
+
+
+class AnimalMetadata(BaseApiAction):
+    """
+    Generate a dictionary of field choices for all animal models.
+    """
+
+    input_model = NoInput
+
+    def experiment_metadata(self):
+        metadata = {}
+        metadata["type"] = tuple_to_dict(models.Experiment.EXPERIMENT_TYPE_CHOICES)
+        return metadata
+
+    def animal_group_metadata(self):
+        metadata = {}
```

Comment:
formatting opinions so there's no "correct", but could we use a `dict` and define in the consturctor, eg,. 

```python
def experiment_metadata(self):
    return dict(
        type=tuple_to_dict(models.Experiment.EXPERIMENT_TYPE_CHOICES),
    )
```